From 83fa6d3ec1ca892e8c0da7099f582281c93f982a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sun, 18 Oct 2026 07:39:15 +0200
Subject: [PATCH] command: Fix settings binding initialization

G_SETTINGS_BIND_GET_NO_CHANGES is intended to be used together
with G_SETTINGS_BIND_GET. Add the latter to ensure preference
widgets are initialized from GSettings without subsequent
settings changes overwriting their values.

This fixes the problem that the command was updated while typing,
not just on focus-out event.

Reference:
https://docs.gtk.org/gio/flags.SettingsBindFlags.html#get_no_changes
---
 command/src/command.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/command/src/command.c b/command/src/command.c
index 8df976304..5111c0065 100644
--- a/command/src/command.c
+++ b/command/src/command.c
@@ -230,9 +230,9 @@ command_settings_callback (GtkAction *action, CommandApplet *command_applet)
 
     dialog = GET_DIALOG ("preferences_dialog");
 
-    g_settings_bind (command_applet->settings, COMMAND_KEY, GET_WIDGET ("command_entry"), "text", G_SETTINGS_BIND_GET_NO_CHANGES);
-    g_settings_bind (command_applet->settings, INTERVAL_KEY, GET_WIDGET ("interval_spinbutton"), "value", G_SETTINGS_BIND_GET_NO_CHANGES);
-    g_settings_bind (command_applet->settings, WIDTH_KEY, GET_WIDGET ("width_spinbutton"), "value", G_SETTINGS_BIND_GET_NO_CHANGES);
+    g_settings_bind (command_applet->settings, COMMAND_KEY, GET_WIDGET ("command_entry"), "text", G_SETTINGS_BIND_GET | G_SETTINGS_BIND_GET_NO_CHANGES);
+    g_settings_bind (command_applet->settings, INTERVAL_KEY, GET_WIDGET ("interval_spinbutton"), "value", G_SETTINGS_BIND_GET | G_SETTINGS_BIND_GET_NO_CHANGES);
+    g_settings_bind (command_applet->settings, WIDTH_KEY, GET_WIDGET ("width_spinbutton"), "value", G_SETTINGS_BIND_GET | G_SETTINGS_BIND_GET_NO_CHANGES);
     g_settings_bind (command_applet->settings, SHOW_ICON_KEY, GET_WIDGET ("show_icon_check"), "active", G_SETTINGS_BIND_DEFAULT);
 
     /* signals */
